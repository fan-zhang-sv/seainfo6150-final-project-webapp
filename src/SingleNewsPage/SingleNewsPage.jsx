import React from "react";
import PropTypes from "prop-types";
import styles from "./SingleNewsPage.module.css";

const SingleNewsPage = (props) => {
  //   var data = decodeURIComponent(escape(window.atob(props.data)));
  var data = decodeURIComponent(props.data);
  data = data.toString().replace(/~~pct~~/g, "%");
  data = JSON.parse(data);
  var { name, provider, url, imgUrl, description } = data;

  return (
    <div className={styles.Container}>
      <div className={styles.Content}>
        <h1>{name}</h1>
        <h2>{provider}</h2>
        {imgUrl !== "" ? (
          <img
            className={styles.Image}
            src={imgUrl}
            alt={name}
            referrerPolicy="no-referrer"
          />
        ) : null}
        <p>{description}</p>
        <a href={url} target="_blank" rel="noopener noreferrer">
          Read On Orginal Page
        </a>
      </div>
    </div>
  );
};

SingleNewsPage.propTypes = {
  data: PropTypes.string.isRequired,
};

export default SingleNewsPage;
