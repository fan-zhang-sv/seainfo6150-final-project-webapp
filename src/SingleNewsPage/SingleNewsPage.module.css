.Container {
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
}

.Content {
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;
}

.Container a {
  font-family: "DM Serif Display", serif;
  font-size: 1.5rem;
  color: black;
}

.Image {
  width: 100%;
  height: 300px;
  object-fit: cover;
}

/* Small devices (landscape phones, 576px and up) */
@media screen and (min-width: 576px) {
  .Image {
    height: 400px;
  }
}

/* Medium devices (tablets, 768px and up) */
@media screen and (min-width: 768px) {
  .Image {
    height: 500px;
  }
  .Content {
    width: 80%;
  }
}

@media screen and (min-width: 992px) {
  .Image {
    height: 600px;
  }
}
