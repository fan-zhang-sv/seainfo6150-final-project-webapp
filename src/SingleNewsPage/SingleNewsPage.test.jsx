import React from "react";
import SingleNewsPage from "./SingleNewsPage";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

const news = {
  name: "FDA authorizes 1st COVID-19 vaccine in United States ",
  url:
    "https://www.msn.com/en-us/health/medical/fda-authorizes-1st-covid-19-vaccine-in-united-states/ar-BB1bRB1S",
  image: {
    thumbnail: {
      contentUrl:
        "https://www.bing.com/th?id=OVFT.-pPAb7rjkmMssGiyXjXE0S&pid=News",
      width: 1200,
      height: 630,
    },
    isLicensed: true,
  },
  description:
    "The Food and Drug Administration authorized the United States' first COVID-19 vaccine for people age 16 and over late Friday, a historic moment in a pandemic that has killed over 290,000 in this",
  provider: [
    {
      _type: "Organization",
      name: "ABC News",
      image: {
        thumbnail: {
          contentUrl:
            "https://www.bing.com/th?id=AR_0cfcddaa71cae000b74e3bf52bc84d78",
        },
      },
    },
  ],
  datePublished: "2020-12-12T04:29:00.0000000Z",
};

describe("SingleNewsPage tests", () => {
  var data = {
    name: news["name"],
    provider: news["provider"][0]["name"],
    url: news["url"],
    imgUrl:
      news["image"] && news["image"]["contentUrl"]
        ? news["image"]["contentUrl"]
        : "",
    description: news["description"],
  };
  data = JSON.stringify(data);
  data = data.toString().replace(/%/g, "~~pct~~");
  data = encodeURIComponent(data);
  it("renders correctly", () => {
    const { container } = render(
      <BrowserRouter>
        <SingleNewsPage data={data} />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
