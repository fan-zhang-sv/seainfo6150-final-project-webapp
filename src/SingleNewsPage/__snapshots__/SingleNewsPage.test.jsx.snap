// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleNewsPage tests renders correctly 1`] = `
<div>
  <div
    class="Container"
  >
    <div
      class="Content"
    >
      <h1>
        FDA authorizes 1st COVID-19 vaccine in United States 
      </h1>
      <h2>
        ABC News
      </h2>
      <p>
        The Food and Drug Administration authorized the United States' first COVID-19 vaccine for people age 16 and over late Friday, a historic moment in a pandemic that has killed over 290,000 in this
      </p>
      <a
        href="https://www.msn.com/en-us/health/medical/fda-authorizes-1st-covid-19-vaccine-in-united-states/ar-BB1bRB1S"
        rel="noopener noreferrer"
        target="_blank"
      >
        Read On Orginal Page
      </a>
    </div>
  </div>
</div>
`;
